### PR TITLE
Switch the prelude to using 'export import'.

### PR DESCRIPTION
**Pull request by jonmeow**
Note, I assume this doesn't affect all the TODOs (like for i32.carbon), but does cut some things down (and requires updating some tests that have prelude name conflicts, but I think they were intended to be updated this way).

### CODE REVIEW

**Comment by jonmeow** [comment]
Depends on #3973

File: toolchain/check/testdata/function/builtin/call_from_operator.carbon [hunk 1]
```diff
@@ -24,53 +14,6 @@ impl i32 as Core.Add {
 
 var arr: [i32; 1 + 2] = (3, 4, 3 + 4);
```
Comment (zygoloid) [inline toolchain/check/testdata/function/builtin/call_from_operator.carbon]: Given that this is a low-level test for calling a builtin via an operator interface, I think we should make it a no-prelude test.

File: toolchain/check/testdata/index/fail_negative_indexing.carbon [hunk 2]
```diff
@@ -6,11 +6,7 @@
 
 var a: (i32, i32) = (12, 6);
 // TODO: Only diagnose this once, and make the diagnostic clearer.
```
Comment (zygoloid) [inline toolchain/check/testdata/index/fail_negative_indexing.carbon]: This TODO no longer applies here. (The underlying issue hasn't been addressed, but we have a `context.TODO` call to track that.)
```suggestion
```

**Review by zygoloid — APPROVED** [review]

File: toolchain/check/testdata/function/builtin/call_from_operator.carbon (hunk 1 above)
Comment (jonmeow) [reply toolchain/check/testdata/function/builtin/call_from_operator.carbon]: Done

File: toolchain/check/testdata/index/fail_negative_indexing.carbon (hunk 2 above)
Comment (jonmeow) [reply toolchain/check/testdata/index/fail_negative_indexing.carbon]: Done

**Pull request merged**